### PR TITLE
Return the trashed status of delete posts/comments

### PR DESCRIPTION
**Pull request by pento**
When a post or a comment is deleted, return a flag to say whether it's been trashed or properly deleted.

See #1260.

### CODE REVIEW

**Comment by pento** [comment]
Okay, I give up on Scrutinizer. Let's commit it.

**Comment by pento** [comment]
Ready for review @WP-API/amigos.

File: lib/endpoints/class-wp-rest-comments-controller.php [hunk 1]
```diff
@@ -282,8 +282,18 @@ public function delete_item( $request ) {
 			}
 
 			$result = wp_trash_comment( $comment->comment_ID );
+			$status = 'trashed';
 		}
 
+		$response = $this->prepare_item_for_response( $comment, $get_request );
```
Comment (rmccue) [inline lib/endpoints/class-wp-rest-comments-controller.php]: If we do this _after_ the comment has been deleted, I think it's possible some preparation might fail here?

Comment (pento) [inline lib/endpoints/class-wp-rest-comments-controller.php]: Good point, moved back to its original position.

**Comment by rachelbaker** [comment]
Merged #1499

**Pull request merged**